api/src/main/java/io/strimzi/api/kafka/model/KafkaMirrorMakerSpec.java

```diff
@@ -0,0 +1,157 @@
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import io.fabric8.kubernetes.api.model.Affinity;
+import io.fabric8.kubernetes.api.model.Toleration;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.strimzi.crdgenerator.annotations.KubeLink;
+import io.strimzi.crdgenerator.annotations.Minimum;
+import io.sundr.builder.annotations.Buildable;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Buildable(
+        editableEnabled = false,
+        generateBuilderPackage = false,
+        builderPackage = "io.fabric8.kubernetes.api.builder"
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({"replicas", "whitelist", "consumer", "producer", "logging", "metrics"})
+public class KafkaMirrorMakerSpec implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    public static final String DEFAULT_IMAGE =
+            System.getenv().getOrDefault("STRIMZI_DEFAULT_KAFKA_MIRRORMAKER_IMAGE", "strimzi/kafka-mirror-maker:latest");
+
+    private int replicas;
+
+    private String whitelist;
+
+    private KafkaMirrorMakerConsumerSpec consumer;
+
+    private KafkaMirrorMakerProducerSpec producer;
+
+    private Resources resources;
+
+    private Affinity affinity;
+
+    private List<Toleration> tolerations;
+
+    private JvmOptions jvmOptions;
+
+    private Logging logging;
+
+    private Map<String, Object> metrics = new HashMap<>(0);
+
+    @Description("The number of pods in the `Deployment`.")
+    @Minimum(1)
+    @JsonProperty(required = true)
+    public int getReplicas() {
+        return replicas;
+    }
+
+    public void setReplicas(int replicas) {
+        this.replicas = replicas;
+    }
+
+    @Description("List of topics which are included for mirroring. This option allows any regular expression using Java-style regular expressions." +
+            "Mirroring two topics named A and B can be achieved by using `--whitelist 'A|B'`. Or you could mirror all topics using `--whitelist '*'`.")
+    public String getWhitelist() {
+        return whitelist;
+    }
+
+    public void setWhitelist(String whitelist) {
+        this.whitelist = whitelist;
+    }
+
+    @Description("Configuration of source cluster.")
+    public KafkaMirrorMakerConsumerSpec getConsumer() {
+        return consumer;
+    }
+
+    public void setConsumer(KafkaMirrorMakerConsumerSpec consumer) {
+        this.consumer = consumer;
+    }
+
+    @Description("Configuration of target cluster.")
+    public KafkaMirrorMakerProducerSpec getProducer() {
+        return producer;
+    }
+
+    public void setProducer(KafkaMirrorMakerProducerSpec producer) {
+        this.producer = producer;
+    }
+
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    @Description("The Prometheus JMX Exporter configuration. " +
+            "See https://github.com/prometheus/jmx_exporter for details of the structure of this configuration.")
```

Comment:
I wonder if we should use an asciidoc attribute for this, which would make it easier to change in the future if that was necessary.